src/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.java

```diff
@@ -18,131 +26,151 @@
     }
 
-    if (scaleType != null) {
-      view.setScaleType(scaleType);
-      scaleType = null;
+    public void setEnableMergePaths(boolean enableMergePaths) {
+        this.enableMergePaths = enableMergePaths;
     }
 
-    if (imageAssetsFolder != null) {
-      view.setImageAssetsFolder(imageAssetsFolder);
-      imageAssetsFolder = null;
+    public void setColorFilterLayers(ArrayList<Map<String,String>> colorLayers) {
+        this.colorFilters = colorLayers;
     }
 
-    if (enableMergePaths != null) {
-        view.enableMergePathsForKitKatAndAbove(enableMergePaths);
-        enableMergePaths = null;
+    /**
+     * Updates the view with changed fields.
+     * Majority of the properties here are independent so they are has to be reset to null
+     * as soon as view is updated with the value.
+     *
+     * The only exception from this rule is the group of the properties for the animation.
+     * For now this is animationName and cacheStrategy. These two properties are should be set
+     * simultaneously if the dirty flag is set.
+     */
+    public void commitChanges() {
+        LottieAnimationView view = viewWeakReference.get();
+        if (view == null) {
+            return;
+        }
+
+        if (animationJson != null) {
+            view.setAnimation(new JsonReader(new StringReader(animationJson)));
+            animationJson = null;
+        }
+
+        if (animationNameDirty) {
+            view.setAnimation(animationName, cacheStrategy);
+            animationNameDirty = false;
+        }
+
+        if (progress != null) {
+            view.setProgress(progress);
+            progress = null;
+        }
+
+        if (loop != null) {
+            view.setRepeatCount(loop ? LottieDrawable.INFINITE : 0);
+            loop = null;
+        }
+
+        if (speed != null) {
+            view.setSpeed(speed);
+            speed = null;
+        }
+
+        if (useHardwareAcceleration != null) {
+            view.useHardwareAcceleration(useHardwareAcceleration);
+            useHardwareAcceleration = null;
+        }
+
+        if (scaleType != null) {
+            view.setScaleType(scaleType);
+            scaleType = null;
+        }
+
+        if (imageAssetsFolder != null) {
+            view.setImageAssetsFolder(imageAssetsFolder);
+            imageAssetsFolder = null;
+        }
+
+        if (enableMergePaths != null) {
+            view.enableMergePathsForKitKatAndAbove(enableMergePaths);
+            enableMergePaths = null;
+        }
+
+        if (colorFilters != null && colorFilters.size() > 0) {
+            int size = colorFilters.size();
+            for (int i = 0; i < size; i++) {
+                Map<String,String> colorLayer = colorFilters.get(i);
+                String layer = colorLayer.get("layer");
+                String color = colorLayer.get("color");
+
+                SimpleColorFilter filter = new SimpleColorFilter(Color.parseColor(color));
```

Comment:
`Color.pareseColor` is not as good a color parser as `processColor`. It supports limited color names, and the alpha channel has to be specific at the beginning of the string like `#FFFF0000` would be solid red instead of `#FF0000FF`. And 3 and 4 letter strings are not recognized.

Reply:
We're only supporting #XXXXXX at the moment and will build out other support as needed.